system-addon/lib/ActivityStreamPrefs.jsm

```diff
@@ -0,0 +1,58 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+"use strict";
+
+const {utils: Cu} = Components;
+Cu.import("resource://gre/modules/Preferences.jsm");
+Cu.import("resource://gre/modules/Services.jsm");
+
+const ACTIVITY_STREAM_PREF_BRANCH = "browser.newtabpage.activity-stream.";
+
+this.Prefs = class Prefs extends Preferences {
+  constructor(branch = ACTIVITY_STREAM_PREF_BRANCH) {
+    super({branch});
+    this._branchName = branch;
+  }
+  get branchName() {
+    return this._branchName;
+  }
+};
+
+this.DefaultPrefs = class DefaultPrefs {
+  constructor(config, branch = ACTIVITY_STREAM_PREF_BRANCH) {
+    this._config = config;
+    this.branch = Services.prefs.getDefaultBranch(branch);
+  }
+  _setDefaultPref(key, val) {
+    switch (typeof val) {
+      case "boolean":
+        this.branch.setBoolPref(key, val);
+        break;
+      case "number":
+        this.branch.setIntPref(key, val);
+        break;
+      case "string":
+        this.branch.setCharPref(key, val);
+        break;
+    }
+  }
+  init(context) {
+    for (const pref of this._config) {
+      let value;
+      if (pref.values) {
+        value = pref.values[context] || pref.values.default;
+      } else {
+        value = pref.value;
+      }
+      this._setDefaultPref(pref.name, value);
+    }
+  }
+  reset() {
+    for (const pref of this._config) {
+      this.branch.clearUserPref(pref.name);
```

Comment:
Oh hrm. Manual testing seems to show that this works on the default branch. I was confused as I thought these changes on the default branch would only affect the defaults. See idl:
"This method does nothing if this object is a default branch."
https://dxr.mozilla.org/mozilla-central/source/modules/libpref/nsIPrefBranch.idl#234-237